.github/workflows/Build.yml

```diff
@@ -1,61 +1,127 @@
 name: Run build
 
 on:
-    push:
-        branches: [master]
-    pull_request:
-        branches: [master]
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
 
 jobs:
-    test:
-        name: "Build (${{ matrix.os }})"
-        runs-on: ${{ matrix.os }}
-        strategy:
-          matrix:
-            os: [ ubuntu-latest, windows-latest ]
+  scope:
+    name: Set scope
+    runs-on: ubuntu-latest
+    outputs:
+      since: ${{ steps.since.outputs.since }}
+    steps:
+      - id: since
+        if: github.event_name == 'pull_request'
+        run: echo "::set-output name=since::-- --since origin/${{ github.base_ref }}"
```

Comment:
I think meybe it's better to ermove `-- --since` from this output strgin?
I think this part should be part of command you run on step:
Now we have `npm run test ${{ needs.scope.outputs.since }}`
and it's not so clear, what is inside `needs.scope.outputs.since`
If we convert it to `npm run test -- --since ${{ needs.scope.outputs.since }}` it tells you that variable is expected to be a `--since` flag value.

BTW, it's just POV